lib/hls/hls_parser.js

```diff
@@ -3335,6 +3365,11 @@ shaka.hls.HlsParser = class {
       presentationDelay = this.maxTargetDuration_ * delaySegments;
     }
 
+    if (this.startTime_ && this.startTime_ < 0) {
+      presentationDelay = Math.min(-this.startTime_, presentationDelay);
```

Comment:
Should this be
```suggestion
      presentationDelay = Math.max(-this.startTime_, presentationDelay);
```
?

Just plugging in some values to `console.log(Math.min(-start, 3*target))` to check:
| EXT-X-START | EXT-X-TARGETDURATION | presentationDelay |
| --- | --- | --- |
| -4 | 4 | 4 |
| -8 | 4 | 8 |
| -15 | 4 | 12 |
| -100 | 4 | 12 |

So you can't set a start value that pushes you further than 3x target duration. But with max it would mean that you can't set a start value that brings you closer to live edge than 3x target duration:
| EXT-X-START | EXT-X-TARGETDURATION | presentationDelay |
| --- | --- | --- |
| -4 | 4 | 12 |
| -8 | 4 | 12 |
| -15 | 4 | 15 |
| -100 | 4 | 100 |

But maybe I'm misunderstanding (I haven't read the rest of the PR yet so not sure how `presentationDelay` is used yet).

Reply:
According to the [RFC 8216 on EXT-X-START](https://datatracker.ietf.org/doc/html/rfc8216#section-4.3.5.2):

> If the Playlist does not contain the EXT-X-ENDLIST tag, the TIME-
      OFFSET SHOULD NOT be within three target durations of the end of
      the Playlist file.

So if the manifest is formatted correctly, we won't have to worry about the EXT-X-START starting us too close to the live edge. Of course, I suppose it wouldn't hurt to be safe in the case of an invalid manifest.

Reply:
@theodab I think the safety is nice... But I was more making the point that the current `Math.min` approach is not applying the EXT-X-START properly when the magnitude of the negative value is greater than 3x target duration... So in the example above, I would expect the `EXT-X-START:-15` case to produce a presentation delay of 15s, but it gets capped at 12s due to the `EXT-X-TARGETDURATION:4` value.

Reply:
@theodab @theRealRobG The start and the delay are two different concepts, delay limits the seekrange for Live, so if it is the start it is less than the delay we change the delay so that we can start in the defined start. But if it is greater it makes no sense to change our logic, and we should allow to do Seek if the user wishes until the delay.

Reply:
Ah, makes sense, thanks!